File: src/leetcode/root-equals-sum-of-children/root-equals-sum-of-children.js
/**
 * Leetcode #2236
 * Difficulty: Easy
 * https://leetcode.com/problems/root-equals-sum-of-children/
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
export const checkTree = root => root.val === root.left.val + root.right.val;

File: src/leetcode/root-equals-sum-of-children/root-equals-sum-of-children.rs
use std::rc::Rc;
use std::cell::RefCell;
impl Solution {
    pub fn check_tree(root: Option<Rc<RefCell<TreeNode>>>) -> bool {
        let node = root.as_ref().unwrap().borrow();
        let left = node.left.as_ref().unwrap().borrow();
        let right = node.right.as_ref().unwrap().borrow();

        node.val == left.val + right.val
    }
}